### PR TITLE
Chore: Change VariableModel to TypedVariableModel in most places

### PR DESCRIPTION
**Pull request by kaydelaney**
Replaced usage of the now deprecated `VariableModel` with `TypedVariableModel` where possible.

### CODE REVIEW

**Review by aangelisc — COMMENTED** [review]
Azure changes LGTM 😊

**Review by torkelo — COMMENTED** [review]
Is typed variable model the runtime state type or the persisted / schema type ?

**Review by torkelo — APPROVED** [review]

File: public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts [hunk 1]
```diff
@@ -252,12 +244,12 @@ export function createDashboardSceneFromDashboardModel(oldModel: DashboardModel)
   });
 }
 
-export function createSceneVariableFromVariableModel(variable: VariableModel): SceneVariable {
+export function createSceneVariableFromVariableModel(variable: TypedVariableModel): SceneVariable {
```
Comment (torkelo) [inline public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts:247]: @kaydelaney I do not think this change is correct, VariableModel here is the schema model imported from schema package,

Comment (torkelo) [reply public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts:247]: but it looks like we where not using the schema types here anyway before, argh..

**Pull request merged**